src/GraphQL/GraphQLMetadataAttribute.cs

```diff
@@ -1,18 +1,44 @@
 using System;
+using GraphQL.Types;
 using GraphQL.Utilities;
 
 namespace GraphQL
 {
+    /// <summary>
+    /// Allows additional configuration to be applied to a type or field definition.
+    /// </summary>
     [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method | AttributeTargets.Property, AllowMultiple = true)]
     public abstract class GraphQLAttribute : Attribute
     {
+        /// <summary>
+        /// Updates the properties of the specified <see cref="TypeConfig"/> as necessary.
+        /// </summary>
         public virtual void Modify(TypeConfig type)
         {
         }
 
+        /// <summary>
+        /// Updates the properties of the specified <see cref="FieldConfig"/> as necessary.
+        /// </summary>
         public virtual void Modify(FieldConfig field)
         {
         }
+
+        /// <summary>
+        /// Updates the properties of the specified <see cref="IGraphType"/> as necessary.
+        /// </summary>
+        public virtual void Modify(IGraphType graphType)
+        {
+        }
+
+        /// <summary>
+        /// Updates the properties of the specified <see cref="FieldType"/> as necessary.
+        /// </summary>
+        /// <param name="fieldType">The <see cref="FieldType"/> to update.</param>
+        /// <param name="isInputType">Indicates if the graph type containing this field is an input type.</param>
```

Comment:
Remember about design considerations about splitting input-output hierarchies of GraphTypes/FieldTypes.

Reply:
Some day in future :)

Reply:
Do you want to deprecate `GraphQLMetadataAttribute`?

Reply:
I'm considering it.  I think I will finish my other changes and then we can consider deprecating `GraphQLMetadataAttribute` considering the rest of the design changes, adding in whatever other attributes are necessary.

Reply:
For example, the `SchemaBuilder` would need changes to support the .NET `Description` attribute, most likely.  I have not checked yet.

Reply:
ok